File: taskflow-fullstack/backend/src/main/java/com/ruhatkaratas/taskflow/task/service/TaskService.java
package com.ruhatkaratas.taskflow.task.service;

import com.ruhatkaratas.taskflow.task.dto.AssignTaskRequest;
import com.ruhatkaratas.taskflow.task.dto.CreateTaskRequest;
import com.ruhatkaratas.taskflow.task.dto.TaskResponse;
import com.ruhatkaratas.taskflow.task.dto.UpdateTaskStatusRequest;
import java.util.List;

public interface TaskService {

    TaskResponse createTask(CreateTaskRequest request, String currentUserEmail);

    List<TaskResponse> getTasksByProject(Long projectId, String currentUserEmail);

    TaskResponse updateTaskStatus(Long taskId, UpdateTaskStatusRequest request, String currentUserEmail);

    TaskResponse assignTask(Long taskId, AssignTaskRequest request, String currentUserEmail);
}

File: taskflow-fullstack/frontend/src/services/task-service.ts
import { apiRequest } from '../lib/api-client'
import type {
  AssignTaskRequest,
  CreateTaskRequest,
  TaskResponse,
  UpdateTaskStatusRequest,
} from '../types/task'

export const taskService = {
  async getTasksByProject(projectId: number): Promise<TaskResponse[]> {
    return apiRequest<TaskResponse[]>(`/api/v1/tasks?projectId=${projectId}`, {
      method: 'GET',
      authenticated: true,
    })
  },

  async createTask(payload: CreateTaskRequest): Promise<TaskResponse> {
    return apiRequest<TaskResponse>('/api/v1/tasks', {
      method: 'POST',
      authenticated: true,
      body: JSON.stringify(payload),
    })
  },

  async updateTaskStatus(taskId: number, payload: UpdateTaskStatusRequest): Promise<TaskResponse> {
    return apiRequest<TaskResponse>(`/api/v1/tasks/${taskId}/status`, {
      method: 'PATCH',
      authenticated: true,
      body: JSON.stringify(payload),
    })
  },

  async assignTask(taskId: number, payload: AssignTaskRequest): Promise<TaskResponse> {
    return apiRequest<TaskResponse>(`/api/v1/tasks/${taskId}/assignee`, {
      method: 'PATCH',
      authenticated: true,
      body: JSON.stringify(payload),
    })
  },
}
